src/hotspot/share/opto/escape.cpp

```diff
@@ -1047,6 +1297,9 @@ void ConnectionGraph::add_call_node(CallNode* call) {
         es = PointsToNode::GlobalEscape;
       }
       add_java_object(call, es);
+      if (es == PointsToNode::GlobalEscape) {
```

Comment:
No need to check for ArgEscape here?

Reply:
Nope. The node was added in the line above. The only two escape_state possible for this node at this moment are NoEscape or GlobalEscape.